packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3227,14 +3236,14 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   }
 
   @override
-  InheritedWidget inheritFromWidgetOfExactType(Type targetType) {
+  InheritedWidget inheritFromWidgetOfExactType(Type targetType, { Object aspect, InheritedElement target }) {
```

Comment:
We should definitely assert that `target` is an ancestor.

Reply:
Good point, done.